projects/distributed-tracing/src/shared/dashboard/widgets/table/table-widget-renderer.component.ts

```diff
@@ -142,16 +142,24 @@ export class TableWidgetRendererComponent
       ])
     );
 
-    this.filterItems = this.model.getFilterOptions().map(filterOption => ({
-      label: capitalize(filterOption.label),
-      value: filterOption
-    }));
-
     this.viewItems = this.model.getViewOptions().map(viewOption => ({
       label: capitalize(viewOption),
       value: viewOption
     }));
 
+    this.populateStaticControls();
+  }
+
+  private populateStaticControls(): void {
+    /*
+     * CAUTION: If TableWidgetViewToggleModel is used, this.model is not hydrated until after setView() is called,
```

Comment:
`this.model` should always be hydrated (first line of the constructor). It sounds like the issue is the view toggle isn't picking a delegate, by default, and forcing the renderer to do so?

Reply:
Correct. That's exactly what it is, but not sure how to default one in the model without a constructor. Just assigning one on the class property doesn't work.

Reply:
sounds like a case for `ModelOnInit`!

Reply:
Oh, lovely. Didn't know we had that. Looks like the fine gentlemen who built the dashboarding system thought of about everything.

Reply:
you'd be amazed by all the cool features we have sitting on a github shelf somewhere 😉

Reply:
I'm still waiting for the variable within a variable use case to come in handy.